teams: split does not permit parent PE wrap-around

davidozog opened the pull request:
# Summary of changes
This attempts to clarify that no team split "wrap-around" is allowed.

I found it a little tricky to specify, so please feel free to propose new or adjusted text.

I'm relying on #1 to account for the `start` argument description.

Changelog entry proposed here:
https://github.com/kwaters4/specification/pull/6/files

Commit:
teams: split does not permit parent PE wrap-around

Commit:
teams: split does not permit parent PE wrap-around

davidozog commented:
@wokuno @maawad - ready for review (looks like I don't have permission to request directly)

maawad commented on content/shmem_team_split_strided.tex:
```diff
@@ -59,6 +59,17 @@
 relative order with respect to the parent team.
 If a $stride$ value equal to 0 is passed to \FUNC{shmem\_team\_split\_strided},
 then the $size$ argument passed must be 1, or the behavior is undefined.
+A newly created team must only include \acp{PE} whose subsequent parent \ac{PE}
```
```suggestion
When $stride$ is nonzero, a newly created team must only include \acp{PE} whose subsequent parent \ac{PE}
```

maawad reviewed (approved):
Thanks, David, for fixing this. I made a minor suggestion but I might be getting too verbose. Looks good to me either way.

davidozog committed:
Update content/shmem_team_split_strided.tex

Co-authored-by: Muhammad Awad <muhammad.a.awad@gmail.com>

wokuno reviewed (approved):
Looks good. Could you just merge this into the `sec/teams-ctx` branch

davidozog commented:
> Looks good. Could you just merge this into the `sec/teams-ctx` branch

Good catch!  I was apparently working too quickly...  I just updated it - will merge in a moment unless I hear otherwise.

davidozog commented:
Oh, just need 1 more approval to merge.

bobkierski commented on content/shmem_team_split_strided.tex:
```diff
@@ -59,6 +59,17 @@
+For example, the list of \acp{PE} in the parent team should not start at a high
+number and then continue to include \acp{PE} in the lower end of the parent
+team's \ac{PE} range.
+If the triplet provided to \FUNC{shmem\_team\_split\_strided} implies such a
```
I would suggest moving this part of the change to the top (line 62).  The content between 62 and 69 would then clarify what is meant by this.

bobkierski replied:
that would also mean that "implies such a wrap-around" would be changed to "implies a wrap-around".

davidozog replied:
Done!  Can you please review the change to see if I captured it well?  Thanks, @bobkierski

Commit:
Merge branch 'pr/no_team_wraparound' of github.com:davidozog/specification into pr/no_team_wraparound

Commit:
teams: improve the explanation of no wrap-arounds

Commit:
teams: revert change to "start" arg description

The fix to "start" is covered in a different PR:
https://github.com/wokuno/specification/pull/1

bobkierski reviewed (approved):
Se ve bueno para mi

davidozog commented:
@wokuno @maawad @bobkierski - looks like this is ready to merge, but I do not have permission to do it.  Could you please hit the big green button if you think it's ready?

maawad commented:
> @wokuno @maawad @bobkierski - looks like this is ready to merge, but I do not have permission to do it. Could you please hit the big green button if you think it's ready?

Maybe @wokuno can help us. I don't have permission to merge either with this PR and #1  too.
